### PR TITLE
Feature interactive list

### PR DESCRIPTION
**Pull request by pountzas**
- Add 2 new JavaScript files and import them as modules:
  - one will contain methods related to drag / drop and sorting functionality.
  - the other will contain methods related to the status updates (completed: true / false).
- Add event listeners to list items (dragstart, dragover, drop) and to checkbox (change).
- Update items object's values for index and completed keys upon user actions.
- Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import './style.css';
+import updateStorage from './storage';
+import checkStatus from './status';
```
Comment (Davidosky007) [inline src/index.js]: * I suggest you import a javascript file as a module that contains methods related to the drag and drop functionality, this is one of the learning objectives for this project [Please refer to the project requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Add%202%20new,completed%3A%20true%20/%20false). currently you have the method in the `index.js` file

File: src/index.js [hunk 2]
```diff
@@ -11,23 +13,95 @@ const list = [
...
+    }
+  }
+}, false);
+
+const checkboxes = document.querySelectorAll('.checkbox');
+
+for (let i = 0; i < checkboxes.length; i += 1) {
+  checkboxes[i].addEventListener('change', (event, sortList) => {
+    checkStatus(event, sortList);
+  }, false);
```
Comment (Davidosky007) [inline src/index.js]: * I get errors when i try to perform the checkbox, drag and drop functionality
* Also the object's value index is not updated upon user actions (when items are being dragged) [Please refer to project requirement ](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Update%20items%20object%27s%20values%20for%20index%20and%20completed%20keys%20upon%20user%20actions.)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
scrnli-7-17-2021-2-35-24-AM

File: src/status.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import updateStorage from './storage';
+
+function checkStatus(sortList, e) {
```
Comment (Davidosky007) [inline src/status.js]: * I get errors in the console when i try this checkStatus functionality please 🙏 kindly make the necessary adjustments 
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
image

File: src/storage.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+function updateStorage(obj) {
+  for (let i = 0; i < Object.keys(obj).length; i += 1) {
+    const key = Object.keys(obj)[i];
+    const arr = localStorage.getItem(key) ? localStorage.getItem(key).split(',') : [];
+    arr.push(obj[key]);
+    localStorage.setItem(key, arr);
+  }
```
Comment (Davidosky007) [inline src/storage.js:7]: * As stated above the object's value index needs to be updated in the local storage for this is a requirement in this project [See here ](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Store%20the%20updated%20array%20of%20items%20in%20local%20storage%2C%20so%20the%20user%20gets%20the%20correct%20list%20values%20after%20the%20page%20reloads.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @pountzas 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
* Please check the comments below and make the necessary changes 
## _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by voscarmv — APPROVED** [review]
Hi @pountzas , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

However, please take note of the following points:

- Please always try to find a way to fix the linter issues without deactivating the linter. For this project it's not necessary to implement proper code modulation, so your linter disabling is valid. But in the future, please research how to fix your code without linter disabling!

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**